src/promptflow-devkit/tests/sdk_cli_test/e2etests/test_flow_run.py

```diff
@@ -1281,6 +1281,27 @@ def test_eager_flow_run_without_yaml(self, pf):
         # the YAML file will not exist in user's folder
         assert not Path(f"{EAGER_FLOWS_DIR}/simple_without_yaml/flow.flex.yaml").exists()
 
+    def test_flex_flow_run_update_signature(self, pf):
```

Comment:
Should also cover flex flow cases:
- With `flow.flex.yaml`
- Without `flow.flex.yaml`

Actually I would recomend add the check logic to existing test case?
![image](https://github.com/microsoft/promptflow/assets/2572521/f1a604a8-ffe9-4022-bac4-8d400016039e)

![image](https://github.com/microsoft/promptflow/assets/2572521/0f41796b-1b40-455b-8741-29756dccc696)

Reply:
Code path is the same. Directly appending extra check after original tests will bring duplicate code. Used `pytest.mark.parameterize` to merge some related tests.